core/src/main/java/net/sourceforge/jnlp/runtime/JNLPPolicy.java

```diff
@@ -235,13 +234,14 @@ private Policy getPolicyFromUrl(String policyLocation) {
             try {
                 final URI policyUri;
                 if (policyLocation.startsWith("file://")) {
-                    policyUri = new URL(policyLocation).toURI();
+                    final String encoded = URLEncoder.encode(policyLocation, UTF_8.name());
+                    policyUri = new URL(encoded).toURI();
                 } else {
                     policyUri = new URI(policyLocation.replace("\\", "/"));
```

Comment:
should we encode here too?
In general we allow specification of urls in the deployment properties file. Should we encode all Urls read by OWS from the deployment properties file?

Reply:
Was thinking the same but I currently do not have the time to find all usages of URLs in the settings.
For now we do not have any other bug report so I would not change any more code.